charts/lws/templates/certmanager/certificatemetrics.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if .Values.enableCertManager }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  labels:
+    app.kubernetes.io/instance: serving-cert
+    {{- include "lws.labels" . | nindent 4 }}
+  name: lws-metrics-cert
+  namespace: {{ .Release.Namespace }}
+spec:
+  dnsNames:
```

Comment:
We should include a subject (e.g. `commonName`) on our certificates. There is an issue that some libraries might not recognize such self-signed cert as valid. This applies also for the webhook.

Please see https://cert-manager.io/docs/configuration/selfsigned/#certificate-validity for more details.

Reply:
If there is a bug with respect to cert-manager configurations that is happening also in webhooks for some libraries, I'd prefer filing an issue to discuss the details and fix it in a followup PR. This PR focuses on building the foundational steps to configure servicemonitor for TLS. I'll again defer the decision to @kubernetes-sigs/lws-maintainers

Reply:
There is an issue which we run into in the kueue PR, making curl not accepting certificates without the subject. So I think it is better to include the subject it in our certificates as well.

- https://github.com/kubernetes-sigs/kueue/pull/4800#discussion_r2014226523
- https://github.com/kubernetes-sigs/kueue/pull/4800#discussion_r2032151059

Reply:
That sounds reasonable. Updated the code by adding `commonName` for metrics certificates.